src/monattack.cpp

```diff
@@ -1648,6 +1648,76 @@ void mattack::leap(monster *z, int index)
     }
 }
 
+void mattack::impale(monster *z, int index)
+{
+    if( !z->can_act() ) {
+        return;
+    }
+    Creature *target = z->attack_target();
+    if( target == nullptr || rl_dist( z->pos(), target->pos() ) > 1 ) {
+        return;
+    }
+
+    player *foe = dynamic_cast< player* >( target );
+    bool seen = g->u.sees( *z );
+
+    z->reset_special(index); // Reset timer
+    z->moves -= 80;
+    bool uncanny = foe != nullptr && foe->uncanny_dodge();
+    if( uncanny || dodge_check(z, target) ){
+    if( foe != nullptr ) {
+        if( seen ) {
+            auto msg_type = foe == &g->u ? m_warning : m_info;
+            foe->add_msg_player_or_npc( msg_type, _("The %s lunges at you, but you dodge!"),
+                                                _("The %s lunges at <npcname>, but they dodge!"),
+                                            z->name().c_str() );
+        }
+        if( !uncanny ) {
+            foe->practice( "dodge", z->type->melee_skill * 2 );
+            foe->ma_ondodge_effects();
+        }
+    } else if( seen ) {
+        add_msg( _("The %s lunges at %s, but misses!"), z->name().c_str(), target->disp_name().c_str() );
+    }
+    return;
+    }
+    auto hit = bp_torso;
+    int dam = rng(35, 45);
+    dam = target->deal_damage( z, hit, damage_instance( DT_CUT, dam ) ).total_damage();
+    if( dam > 0 && foe != nullptr ) {
+        if( seen ) {
+            auto msg_type = foe == &g->u ? m_bad : m_info;
+            //~ 1$s is monster name, 2$s bodypart in accusative
+            foe->add_msg_player_or_npc( msg_type,
+                                        _("The %1$s impales your torso!"),
+                                        _("The %1$s impales <npcname>'s torso!"),
+                                        z->name().c_str());
+        }
+        foe->practice( "dodge", z->type->melee_skill );
+        if( one_in( 30 / dam ) ) {
+            if( foe->has_effect( "infected", hit ) ) {
+                foe->add_effect( "infected", 50, hit, true );
+                foe->apply_damage( z, bp_torso, dam * 1.5 );
+                foe->moves -= dam * 20;
```

Comment:
I meant the `one_in` that governs infection. Impale either infects and deals extra damage or does neither.

I misread that part about apply damage - looks like apply already ignores armor.

That move drain would be better off either constant or otherwise not dependent on player armor. To hit the tanks harder and the lightly armored survivors lighter (being a tank is already too good, while being lightly armored too weak to be worth it).